A32NX/html_ui/Pages/A32NX_Core/A32NX_TransitionAltitude.js

```diff
@@ -0,0 +1,345 @@
+class A32NX_TransitionAltitude {
+    init() {
+        console.log('A32NX_TransitionAltitude init');
+        this.currentDeparture = "";
+        this.currentArrival = "";
+        this.offline = false;
+        this.checkStart = true;
+        this.checkStartSec = 0;
+        this.testAirport = "";
+        this.trySendRequest();
+    }
+
+    update(_deltaTime, _core) {
+        if (this.offline === true) {
+            this.offlineTransAlt();
+        }
+        this.tryCheckAPI(_deltaTime);
+    }
+
+    async trySendRequest() {
+        await NXApi.getAirport("KLAX")
+            .then((data) => {
+                this.testAirport = data.transAlt;
+            });
+    }
+
+    tryCheckAPI(_deltaTime) {
+        if (this.offline === false && this.checkStart === true) {
+            this.checkStartSec += _deltaTime;
+        }
+        if ((this.checkStartSec >= 60 * 1000) && (this.testAirport === "")) {
+            this.offline = true;
+            this.checkStart = false;
+        } else if ((this.checkStartSec >= 60 * 1000) && (this.testAirport !== "")) {
+            this.offline = false;
+            this.checkStart = false;
+            this.checkStartSec = 0;
+        }
+    }
+
+    offlineTransAlt() {
+        const Departure = NXDataStore.get("PLAN_ORIGIN", "");
+        const Arrival = NXDataStore.get("PLAN_DESTINATION", "");
+        if (Departure !== "" && Arrival !== "") {
+            if (this.currentDeparture !== Departure) {
+                this.departureLogic(Departure);
+                this.currentDeparture = Departure;
+            }
+            if (this.currentArrival !== Arrival) {
+                this.arrivalLogic(Arrival);
+                this.currentArrival = Arrival;
+            }
+        }
+    }
+
+    departureLogic(airport) {
+        const airportDepart_1 = airport.substr(0,1);
+        const airportDepart_2 = airport.substr(0,2);
+        if (airportDepart_1 === "K" || airportDepart_1 === "C") {
+            SimVar.SetSimVarValue("L:AIRLINER_TRANS_ALT", "Number", 18000); // Canada & USA fixed to 18,000ft
+        } else if (airportDepart_1 === "Y") {
+            SimVar.SetSimVarValue("L:AIRLINER_TRANS_ALT", "Number", 10000); // SPA region only have 10,000ft & 11,000ft. But 10,000ft is more so using that
```

Comment:
**thought**: SPA reminds me of a town in Belgium. I'm guessing not everyone is familiar with the abbreviation.
**thought**: sentences usually end with a `.`

Reply:
Ok

Reply:
It's nice you changed the comment to Belgium. However, as you programmed it _you_ probably know which country it relates to. Hint: it's not Belgium, as their ICAO airport code doesn't start with `Y`.

Reply:
oh i see